contracts/bonding/BondingManager.sol

```diff
@@ -524,6 +531,9 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
             if (currPool.cumulativeRewardFactor == 0) {
                 currPool.cumulativeRewardFactor = cumulativeFactorsPool(newDelegate, newDelegate.lastRewardRound)
                     .cumulativeRewardFactor;
+                if (currPool.cumulativeRewardFactor == 0) {
```

Comment:
This was meant to account for the case where the delegator bonds to a transcoder that has never called reward. In that case, the `cumulativeRewardsFactor` of the round where the bond was made would stay zeroed and that would affect the pending rewards calculation on the checkpoints contract.

After thinking a bit further about this, also related to the other things we've talked about, I've decided to undo this change to `BondingManager`. Will be good to reduce the amount of changes here and instead have the checkpointing code resilient to such imperfect states that might exist in the bonding manager.

This is more aligned with that idea of not failing on read transactions as much as possible, assuming a default uninitialized state instead. I'm compiling a commit to fix this issue together with the whole mindset shift to return 0 for un-checkpointed accounts or when querying before their first checkpoint. This would make the contract better to use, which I've learned the hard way while trying to interact with the current implementation on the testnet. Will send a detailed explanation of the bigger changes here after I update the PR!

Reply:
Reverted this change on https://github.com/livepeer/protocol/pull/614/commits/d5062fe2db7d5037988c2f1a5989886d7ce087d3